### PR TITLE
Development

### PR DESCRIPTION
**Pull request by lils-h-lils**
merging into my main code

### CODE REVIEW

**Comment by FrankOneMillion** [comment]
errors are handled well with use of debug.assert
code is written well with correct use of camelCase and PascalCase meeting the guidelines
testing is implemented well testing both the die class methods and game class methods
could implement additional comments in a XML style if you wish to achieve a higher grade :)

File: CMP1903_A1_2324/Program.cs [hunk 1]
```diff
@@ -10,10 +10,18 @@ internal class Program
     {
         static void Main(string[] args)
         {
-            /*
-             * Create a Game object and call its methods.
-             * Create a Testing object to verify the output and operation of the other classes.
-             */
+            //Create a Game object and call its methods 
+            Game game = new Game();
+            int total = game.RollDice();
```
Comment (ExactlyTwelveSharks) [inline CMP1903_A1_2324/Program.cs:20]: 'int total = game.RollDice();'
By calling the RollDice function in the constructor of the Game class, you could remove this step.

**Comment by ExactlyTwelveSharks** [comment]
1. Have the requirements been met?
The requirements have been met and you have implemented the the testing class using the specified debug.assert. Good job!

2. Is the code formatted using the Style Guidelines correctly?
The code follows the style guidelines well with correct use of camel and pascal case for the variables, I can't find any areas where this hasn't been applied correctly.

3. Is the code easy to read?
The code is easy to read, with helpful comments and a good layout. Though I would suggest maybe a bit more commenting to make it a little more understandable.

4. Are different errors handled correctly?
Errors are handled correctly using debug.assert as instructed. Overall, you've done very well.

**Review by FrankOneMillion — COMMENTED** [review]
(*Same as Comment, i put in wrong section*)
errors are handled well with use of debug.assert
code is written well with correct use of camelCase and PascalCase meeting the guidelines
testing is implemented well testing both the die class methods and game class methods
could implement additional comments in a XML style if you wish to achieve a higher grade :)

**Pull request merged**